src/blockdata/constants.rs

```diff
@@ -129,18 +149,24 @@ pub fn genesis_block(network: Network) -> Block {
             }
         }
         Network::Signet => {
-            // TODO: somehow obtain the block script and generate coinbase tx from it
-            let txdata = vec![bitcoin_genesis_tx()];
+            let block_script = script::Builder::new().push_opcode(opcodes::all::OP_PUSHNUM_1)
+                                                     .push_slice(&Vec::from_hex("03ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430").unwrap())
+                                                     .push_opcode(opcodes::all::OP_PUSHNUM_1)
+                                                     .push_opcode(opcodes::all::OP_CHECKMULTISIG)
+                                                     .into_script();
+            let signet_txdata = vec![bitcoin_signet_genesis_tx(block_script)];
+            let signet_hash: sha256d::Hash = signet_txdata[0].txid().into();
+            let signet_merkle_root = signet_hash.into();
             Block {
                 header: BlockHeader {
                     version: 1,
                     prev_blockhash: Default::default(),
-                    merkle_root,
+                    merkle_root: signet_merkle_root,
```

Comment:
I think this can be `signet_txdata[0].txid().as_hash().into()` so you can avoid the 2 lines of conversion above.

Reply:
Sounds good